### PR TITLE
Parametrize pytests

### PR DESCRIPTION
**Pull request by becantwell**
Added `pytest_generate_tests` functionality and a new scenario type fixture to parameterize tests by test application and scenario type

### CODE REVIEW

**Comment by Mani-D** [comment]
The [ci ](https://travis-ci.org/Morgan-Stanley/Xpedite/builds/473939967?utm_source=github_status&utm_medium=notification) doesn't show a pytest for each scenario/app, instead we still get the old behavior .  Can you please check ?

File: test/pytest/conftest.py [hunk 1]
```diff
@@ -19,46 +23,66 @@ def pytest_addoption(parser):
...
 def apps(request):
   """
-  Directory to extract files to
+  Apps to test
   """
   apps = request.config.option.apps
   return apps.split(',')
+
+@pytest.fixture(scope='module')
+def scenarios(request):
```
Comment (Mani-D) [inline test/pytest/conftest.py]: rename scenarios to scenarioTypes

File: .travis.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ script:
   - ./build.sh
   - ./install.sh
   - ./test/validateTarFiles.sh
-  - ./test/runTest.sh -c
+  - ./test/runTest.sh -c -a allocatorApp,dataTxnApp,multiThreadedApp,slowFixDecoderApp
```
Comment (Mani-D) [inline .travis.yml]: make this the default to run test for all apps, instead of just slow fix decoder ?

File: test/pytest/conftest.py (hunk 1 above)
Comment (becantwell) [reply test/pytest/conftest.py]: Made this change, it's instead of enums right?

**Pull request merged**